### ISSUE TITLE
extract_area: bad extract area - Error report for exile-diary 1.2.0

### ISSUE THREAD

**Issue by ptrang** [Qt-dev/exile-diary#213]
Error:
```
Error: extract_area: bad extract area
```
OS: Windows NT 10.0.22621
App: exile-diary 1.2.0

using steam screenshots.  (cannot use f8 because path of exile makes blank/black screenshots)

**Comment by Qt-dev** [comment]
Did you try the new screenshot mode?
If you go in the settings, 1.2 should have an option to take screenshots using CTRL+F8 instead of reading screenshot files:

![image](https://github.com/Qt-dev/exile-diary/assets/3586102/f3bf8aa6-3c79-42c7-bc7c-6a3a2b179233)

It uses something else than the normal screenshot generation and just takes a snap of the display of the window, so there should be no black background and it should work a bit better if you can try that out.

If you want to look at what the captures look like, they will all be in `%APPDATA%\exile-diary\.dev_captures`.

Let me know, and don't hesitate to add the screenshots from that folder so that I can take a look at what went wrong there :)

**Comment by ptrang** [comment]
![20231009065211_1](https://github.com/Qt-dev/exile-diary/assets/1136436/5e29367f-ba4e-4bc4-96ed-82c285ab950c)
![20231009072424_1](https://github.com/Qt-dev/exile-diary/assets/1136436/f1742f37-9feb-4411-8be3-121a2d171a23)
![20231009080731_1](https://github.com/Qt-dev/exile-diary/assets/1136436/d4746edc-0f78-40e1-b3b9-77e49278d6cd)
attached a few pics, i'll try the custom screenshot shortcut

**Comment by Qt-dev** [comment]
Thank you for these. They look like they should be ok, but the screenshots in `.dev_captures` should make it more obvious what went wrong

**Comment by ptrang** [comment]
Custom screenshot button has been working smoothly for me. 

Would dev_captures show what is being pulled from the steam screenshot folder?  Is there a chance that the software is processing pictures from the thumbnails folder that is under the steam screenshots folder at the same time as the regular screenshot?

**Comment by Qt-dev** [comment]
That's possible. It's setup to listen to the folder but it might be listening to any subfolder too.
It also might be that the format is slightly different in the way it handles colors or optimization.

I am not looking at the code right now but I believe the dev_captures directory should include all the processing pictures (full shot, split screenshot, and what it found as the area stats and the mods box)

It replaces it on every try but you could take a screenshot with Steam and check if these are generated.
In theory, it should have the full shot and the split screenshot

**Comment by ptrang** [comment]
I kept an eye on the captures directory while I took a steam screenshot, and the 4 pics look fine to me (below pics look based on the steam screenshot).  I can't tell based on logs if it's opening the thumbnail pic, but can't think of any other reason why it's saying bad extract area

![cropped-screenshot](https://github.com/Qt-dev/exile-diary/assets/1136436/77a7de91-554a-4690-aa5a-1c7574fb3174)
![mods](https://github.com/Qt-dev/exile-diary/assets/1136436/f13cbe78-d548-4e33-be4d-a13e45074162)
![screenshot](https://github.com/Qt-dev/exile-diary/assets/1136436/21d2b2b1-3c9f-4280-a935-6318bb1f1fad)
![stats](https://github.com/Qt-dev/exile-diary/assets/1136436/38d7d12c-0c67-4f1d-8759-6da326617bf1)

**Comment by Qt-dev** [comment]
Right, these look perfect indeed. I think you're right, it must be reading thumbnails and erroring on these if they are in subfolders.

You can verify that by checking the logs of exile diary, you should have a `Cropping Screenshot YOUR_THUMBNAIL` line in there before the error.

**Comment by ptrang** [comment]
ah indeed, i should have looked for that log file first. could have saved us quite a bit of conversation haha
[2023-10-09 15:26:45.275] [info]  (main-screenshot-watcher) Cropping new screenshot: C:\Program Files (x86)\Steam\userdata\1075234\760\remote\238960\screenshots\thumbnails\20231009152643_1.jpg
[2023-10-09 15:26:45.276] [info]  (main-screenshot-watcher) Cropping new screenshot: C:\Program Files (x86)\Steam\userdata\1075234\760\remote\238960\screenshots\20231009152643_1.jpg
[2023-10-09 15:26:45.276] [info]  (main-screenshot-watcher) Cropping new screenshot: C:\Program Files (x86)\Steam\userdata\1075234\760\remote\238960\screenshots\20231009152643_1.jpg
[2023-10-09 15:26:45.276] [info]  (main-screenshot-watcher) Cropping new screenshot: C:\Program Files (x86)\Steam\userdata\1075234\760\remote\238960\screenshots\thumbnails\20231009152643_1.jpg

**Comment by Qt-dev** [comment]
Haha all good, it's good info on stuff I am not necessarily familiar with :)

Looks like you're right, the thumbnail is getting caught by the watcher and then it cannot find the right dimensions to split the mods and the area info at the top.

For the fact that it is cropping it twice, I do not think that's an issue in your case, but I think I see where the issue is. When you changed settings, it registered a listener again, and so you have 2 processes that are catching any new file in there.
Changing the settings again will add a new one again, but then restarting the app altogether will revert this back to normal :)

I created #214 and #215 to track cleaning that up in a next version.
Disabling screenshot reading and relying on that new screenshot stuff should work for you for now.
